pages/docs/tutorials/native/dynamic-libraries.md

```diff
@@ -0,0 +1,339 @@
+the `kotlin.root.demo` anonymous struct.
+ 
+### The Entry Point
+
+We see how API is created. To start with, we need to initialize the 
+`demo_ExportedSymbols` structure. Let's take a look at the latest part 
+of the `demo_api.h` for that:
+
+```c
+extern demo_ExportedSymbols* demo_symbols(void);
+```
+
+The function `demo_symbols` allows us to open the door from a native 
+code to Kotlin/Native library. That is the entry point we use. The 
+library name is used as the prefix for the function name. 
+
+
+Kotlin/Native object references does not support multi-threaded access. It
+might be necessary to host the returned `demo_ExportedSymbols*` pointer
+per thread.
+
+## Using Generated Headers from C
+
+The usage from C not complicated and strait forward. We create a `main.c` file with the following 
+code: 
+```c
+#include "demo_api.h"
+#include "stdio.h"
+
+int main(int argc, char** argv) {
+  //initialize Kotlin/Native library
+  demo_ExportedSymbols* lib = demo_symbols();
+
+  //call functions
+  lib->kotlin.root.demo.ints(1, 2, 3, 4);
+  lib->kotlin.root.demo.floats(1.0f, 2.0);
+
+  //use C and Kotlin/Native strings
+  const char* str = "Hello from Native!";
+  const char* response = lib->kotlin.root.demo.strings(str);
+  printf("in: %s\nout:%s\n", str, response);
+  lib->DisposeString(response);
+
+  //create Kotlin object instance
+  demo_kref_demo_DemoClazz newInstance 
+          = lib->kotlin.root.demo.DemoClazz.DemoClazz();
+  long result = lib->kotlin.root.demo.DemoClazz.foo(newInstance, 42);
+  printf("DemoClazz returned %ld\n", result);
+  lib->DisposeStablePointer(newInstance.pinned);
+
+  return 0;
+}
+```
+
+## Compiling and Running the Example on Linux and macOS
+
+On macOS 10.13 with Xcode, we compile the C code and link it with the dynamic library
+with the following command:
+```bash
+gcc main.c libdemo.dylib
+```
+
+On Linux we call the similar command: 
+```bash
+gcc main.c libdemo.so
+```
+
+The compiler generates the executable called `a.out`. We run it to see Kotlin code
+being executed from C library in action. On Linux we'll need to include `.` into the `LD_LIBRARY_PATH`
+to let the application to load the `libdemo.so` library from the current folder.
+
+## Compiling and Running the Example on Windows
+
+To start with, you'll need Microsoft Visual C++ compiler installed and support AMD x64 
+target. The easiest way is to have a version of Microsoft Visual Studio installed on 
+your Windows machine. 
+
+We will be using `x64 Native Tools Command Prompt <VERSION>` console. You'll see the 
+shortcut to open the console in the start menu. It comes with a Microsoft Visual Studio
+package.  
+
+On Windows Dynamic libraries are included either via a special static library wrappers,
+or via a manual code with the [LoadLibrary](https://msdn.microsoft.com/en-us/library/windows/desktop/ms684175.aspx)
+function call. We will follow the first option and generate the static wrapper library
```

Comment:
Please fix all that: we generate .def file, and we could also generate both static and dynamic libraries by K/N.

Reply:
I found the .def file generation with K/N 0.8.2. Cool!. On Windows one needs to have a static library wrapper that uses the DLL. Are we able to generate one? Right now I see no such library generated.